source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -111,22 +194,23 @@ void CompressorFilter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallba
 
 Http::FilterHeadersStatus CompressorFilter::encodeHeaders(Http::ResponseHeaderMap& headers,
                                                           bool end_stream) {
+  const auto& config = config_->responseDirectionConfig();
   const bool isEnabledAndContentLengthBigEnough =
-      config_->enabled() && isMinimumContentLength(headers);
-  const bool isCompressible = isEnabledAndContentLengthBigEnough && isContentTypeAllowed(headers) &&
+      config.compressionEnabled() && config.isMinimumContentLength(headers);
+  const bool isCompressible = isEnabledAndContentLengthBigEnough &&
+                              config.isContentTypeAllowed(headers) &&
                               !hasCacheControlNoTransform(headers) && isEtagAllowed(headers) &&
                               !headers.getInline(content_encoding_handle.handle());
   if (!end_stream && isEnabledAndContentLengthBigEnough && isAcceptEncodingAllowed(headers) &&
```

Comment:
want to hear your thought on the intermediate boolean logic vs. a long && statement as done above for request direction?

Reply:
I considered this possibility, but then decided that the conditions have too little in common and an intermediate function may make the logic less readable after all.

I'll think again :)